node_modules/gh-users/tests/test-import.js

```diff
@@ -0,0 +1,302 @@
+/**
+ * Copyright (c) 2014 "Fronteer LTD"
+ * Grasshopper Event Engine
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+var _ = require('lodash');
+var assert = require('assert');
+var fs = require('fs');
+
+var AuthTestsUtil = require('gh-auth/tests/util');
+var ConfigTestsUtil = require('gh-config/tests/util');
+var RestAPI = require('gh-rest');
+var TestsUtil = require('gh-tests/lib/util');
+
+var UsersTestsUtil = require('./util');
+
+describe('Users - import', function() {
+
+    /**
+     * Test that verifies that users can be imported
+     */
+    it('verify users can be imported', function(callback) {
+        TestsUtil.getGlobalAdminRestClient(function(globalAdminClient) {
+            TestsUtil.generateTestTenant(1, function(tenant, app) {
+
+                // Import some local users
+                var stream = fs.createReadStream(__dirname + '/data/users.csv');
+                UsersTestsUtil.assertImportUsers(globalAdminClient, stream, app.id, 'local', false, function() {
+
+                    // Import some shibboleth users
+                    stream = fs.createReadStream(__dirname + '/data/users-shib.csv');
+                    UsersTestsUtil.assertImportUsers(globalAdminClient, stream, app.id, 'shibboleth', false, function() {
+
+                        // Assert the users were imported
+                        UsersTestsUtil.assertGetUsers(globalAdminClient, app.id, 'Christina McChristina', null, null, function(users) {
+                            var user = _.find(users.results, {'displayName': 'Christina McChristina'});
+                            assert.ok(user);
+                            assert.strictEqual(user.displayName, 'Christina McChristina');
+                            assert.strictEqual(user.email, 'christina@example.com');
+                            UsersTestsUtil.assertUser(user, null, true);
```

Comment:
It might be too late to change this, but a function like this is really hard to read without having all parameters memorized (which is never, for me), because it doesn't indicate anything about what it is asserting. It would be nice if we just had methods that asserted something specific, like:

`assertUserEquals`
`assertUserVisible`